test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -248,123 +262,108 @@ private void CopyLatestCliToTestDirectory(string destinationDir)
                 {
-                    File.Delete(targetFilePath);
+                    FileUtility.Replace(
+                        sourceFileName: Path.Combine(packProjectCoreArtifactsDirectory.FullName, "ilmerge", packFileName),
+                        destFileName: Path.Combine(packAssemblyDestinationDirectory, packFileName));
                 }
-                catch
+                else
                 {
-                    // Do nothing
-                }
-                using (var stream = entry.Open())
-                {
-                    var copiedFile = packageFileExtractor.ExtractPackageFile(packageFileName, targetFilePath, stream);
-                    if (copiedFile != null)
+                    foreach (var assembly in packProjectCoreArtifactsDirectory.EnumerateFiles("*.dll"))
                     {
-                        entry.UpdateFileTimeFromEntry(copiedFile, logger);
-
-                        filesCopied.Add(copiedFile);
+                        OverwriteFile(
+                            sourceFileName: assembly.FullName,
+                            destFileName: Path.Combine(packAssemblyDestinationDirectory, assembly.Name));
                     }
                 }
+                // Copy the pack targets
+                var packTargetsSource = Path.Combine(packProjectCoreArtifactsDirectory.FullName, packTargetsName);
+                var targetsDestination = Path.Combine(pathToPackSdk, "build", packTargetsName);
+                var targetsDestinationCrossTargeting = Path.Combine(pathToPackSdk, "buildCrossTargeting", packTargetsName);
+                OverwriteFile(packTargetsSource, targetsDestination);
+                OverwriteFile(packTargetsSource, targetsDestinationCrossTargeting);
             }
-
         }
 
-        private void CopyNupkgFilesToTarget(PackageArchiveReader nupkg, string destPath, IEnumerable<string> files)
+        private void OverwriteFile(string sourceFileName, string destFileName)
         {
-            var packageFileExtractor = new PackageFileExtractor(files,
-                                         PackageExtractionBehavior.XmlDocFileSaveMode);
-
-            nupkg.CopyFiles(destPath, files, packageFileExtractor.ExtractPackageFile, new TestCommandOutputLogger(),
-                CancellationToken.None);
-
-        }
-
-        private static string FindMostRecentNupkg(string nupkgDirectory, string id)
-        {
-            var info = LocalFolderUtility.GetPackagesV2(nupkgDirectory, new TestLogger());
-
-            return info.Where(t => t.Identity.Id.Equals(id, StringComparison.OrdinalIgnoreCase))
-                .Where(t => !Path.GetExtension(Path.GetFileNameWithoutExtension(t.Path)).Equals(".symbols"))
-                .OrderByDescending(p => p.LastWriteTimeUtc)
-                .First().Path;
+            FileUtility.Delete(destFileName);
+            File.Copy(sourceFileName, destFileName);
```

Comment:
I think we can simplify this code by calling [this](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.copy?view=netframework-4.8#System_IO_File_Copy_System_String_System_String_System_Boolean_) overload of `File.Copy`